src/gd_gif_in.c

```diff
@@ -315,8 +315,9 @@ BGD_DECLARE(gdImagePtr) gdImageCreateFromGifCtx(gdIOCtxPtr fd)
 			ReadImage(im, fd, width, height, localColorMap, BitSet(buf[8], INTERLACE), &ZeroDataBlock);
 		} else {
 			if(!haveGlobalColormap) {
-				gdImageDestroy(im);
-				return 0;
+				ColorMap[CM_RED][1] = 0xff;	//Still a valid gif, apply simple default palette as per spec
```

Comment:
can you move the comment above the line instead of being inline ?

also put a space after the `//`

Reply:
yup sure